src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -280,29 +278,7 @@ public ResponseEntity<User> registerForCourse(@PathVariable Long courseId) {
         Course course = courseRepository.findWithEagerOrganizationsElseThrow(courseId);
         User user = userRepository.getUserWithGroupsAndAuthoritiesAndOrganizations();
         log.debug("REST request to register {} for Course {}", user.getName(), course.getTitle());
-        if (allowedCourseRegistrationUsernamePattern.isPresent() && !allowedCourseRegistrationUsernamePattern.get().matcher(user.getLogin()).matches()) {
-            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(applicationName, false, Course.ENTITY_NAME, "registrationNotAllowed",
```

Comment:
this is more of a question and not a suggestion: now instead of this, a new AccessForbiddenException("Registration with this username is not allowed."); is thrown. Do you know if there is any significant difference between this (if we were to change .badRequest() to .status(HttpStatus.FORBIDDEN)) and the new line (throw new AccessForbiddenException("Registration with this username is not allowed.");) ?

Reply:
I'm not an expert on this, but I believe there should not be any difference except that the exception can more easily propagate through the code => The information that access is not allowed does not have to be passed through all layers this way